### DISCUSSION TITLE
Paywall display modes

### DISCUSSION THREAD

**Discussion by askalia** [Glench/ExtPay#138] [Ideas]
Currently the paywall is only displayed as a popup. no choice.
popup thats displayed is either as a fullpage or a popup that comes with the risk to be hidden behind the browser window.
Moreover more than one popup can be displayed and overlap each other, instead of one single popup instance always on top.
This lead to a poor experience.

I think of other display modes : 
- open another tab and make it active
- embed in the page where the content-script is live : iframe.

**Comment by burkybang** [comment]
If you know JavaScript and web extension API, you can do that by picking apart the ExtensionPay JS file. I see the ExtensionPay JS file as a suggestion and an example of how to use the API.

**Reply by askalia** [reply]
I can do this, I inspected the JS file but would be great to have it native to the lib.
Otherwise extpay get less useful.

**Reply by burkybang** [reply]
I don't feel that way at all. The JS file is such a small part of ExtensionPay.

**Reply by Glench** [reply]
Only the ExtPay.js file is officially supported. If you use ExtensionPay in a non-official way then you are taking a risk that your extension will no longer work.
